src/sentry/mediators/sentry_apps/updater.py

```diff
@@ -49,6 +52,24 @@ def call(self):
         self.sentry_app.save()
         return self.sentry_app
 
+    @if_param("features")
+    def _update_features(self):
+        if not self.user.is_superuser and self.sentry_app.status == SentryAppStatus.PUBLISHED:
+            raise APIError("Cannot update features on a published integration.")
+        # Delete any unused features
+        unused_features = IntegrationFeature.objects.filter(
+            target_id=self.sentry_app.id, target_type=IntegrationTypes.SENTRY_APP.value
+        ).exclude(feature__in=list(self.features))
+        for unused_feature in unused_features:
+            unused_feature.delete()
+        # Create any new features
+        for feature in self.features:
+            IntegrationFeature.objects.get_or_create(
+                target_id=self.sentry_app.id,
+                target_type=IntegrationTypes.SENTRY_APP.value,
+                feature=feature,
+            )
```

Comment:
Since this is exactly what we're doing for `doc_integrations` and we will probably reuse this in the future, it might be more effective to throw this operation into its own manager method:

In this file:
```python
IntegrationFeature.objects.clean_update(target_id=self.sentry_app.id, target_type=IntegrationTypes.SENTRY_APP)
```

In `integrationfeature.py`:
```python
class IntegrationFeatureManager(BaseManager):
    ...
    def clean_update(
        incoming_features: List[int],
        target: Union[SentryApp, DocIntegration],
        target_type: IntegrationTypes,
    ):
        # Delete any unused features
        unused_features = IntegrationFeature.objects.filter(
            target_id=target.id, target_type=target_type.value
        ).exclude(feature__in=incoming_features)
        for unused_feature in unused_features:
            unused_feature.delete()
        # Create any new features
        for feature in incoming_features:
            IntegrationFeature.objects.get_or_create(
                target_id=target.id,
                target_type=target_type.value,
                feature=feature,
            )
```